holoviews/core/data/util.py

```diff
@@ -60,7 +60,7 @@ def cached(method):
     Decorates an Interface method and using a cached version
     """
     def cached(*args, **kwargs):
-        cache = getattr(args[1], '_cached')
+        cache = args[1]._cached
```

Comment:
Your suggested change will change the behavior of the original code, which I don't want to do in this PR.

Reply:
But don't you think that was the original intention of the code? That'd be my guess given the line below. Merging this change sorts of cement the bug, if that's a bug (it's easy to be forget that `getattr` errors out when the attribute doesn't exist instead of returning `None` :( ).

Reply:
I agree with you. However, on top of that it seems to me that the intention of bugbear is not just to refactor code:

> A plugin for flake8 finding likely bugs and design problems in your program

The "likely" is important to me, applying these changes without looking into them any further means we don't know whether bugbear spotted a real bug or not.

In this case bugbear did the job well, it spotted a line of code that could potentially hide a bug. It doesn't know the proper way to fix it though, the suggestion it makes is purely a no-op (probably bugbear has a mode to just emit warnings and no change?).

Reply:
I haven't enabled this rule, so it will not run automatically with pre-commit. I have only run it manually with the command in the first post. I don't plan on enabling it in this PR.

> It doesn't know the proper way to fix it though

I'm afraid I have to disagree that it does not correctly fix it. The change made does not change the behavior of the code. A reason for enabling something like this is that if the original author had seen this fix made and disagreed, he could have taken action directly. That is no longer the case, and we don't know what the original author thought, and even if we did, someone could depend on this behavior.  

Ruff only auto-fixes if the change is 1-to-1. A warning will be raised if you break the rule, and the fix needed can or will change the behavior. I looked through all the changes manually and did some myself because it only had a warning.

Reply:
> A reason for enabling something like this is that if the original author had seen this fix made and disagreed, he could have taken action directly.

I 100% agree with that!  As you say this would have worked perfectly fine had this been set up at the time this code was committed. Now it's too late, we don't know what change should be made, why do we need to make any change at all? :)

Reply:
Looking at the code, I think it is correct to raise an error if `_cached` is not set. It is explicit set to `None` [here](https://github.com/holoviz/holoviews/blob/main/holoviews/core/data/__init__.py/#L354).

If it does not raise an error, it will never cache anything (without you knowing) and therefore defeat the purpose of the function.

Reply:
Ok thanks for checking!